src/syscheckd/syscheck_audit.c

```diff
@@ -652,10 +652,14 @@ void audit_parse(char *buffer) {
             // user_name & user_id
             if(regexec(&regexCompiled_uid, buffer, 2, match, 0) == 0) {
                 match_size = match[1].rm_eo - match[1].rm_so;
-                os_malloc(match_size + 1, w_evt->user_id);
-                snprintf (w_evt->user_id, match_size +1, "%.*s", match_size, buffer + match[1].rm_so);
-                const char *user = get_user("", atoi(w_evt->user_id), NULL);
+                char *uid = NULL;
+                os_malloc(match_size + 1, uid);
+                snprintf (uid, match_size +1, "%.*s", match_size, buffer + match[1].rm_so);
+                char *user = get_user("",atoi(uid), NULL);
                 w_evt->user_name = strdup(user);
+                w_evt->user_id = strdup(uid);
```

Comment:
Calls to `strdup` function are unnecessary and messing the code. Please delete them.

Also, note that we should use `os_strdup` instead of `strdup` to check for errors.